libs/client-sdk/src/client_impl.cpp

```diff
@@ -112,14 +110,15 @@ ClientImpl::ClientImpl(Config config,
     flag_manager_.LoadCache(context_);
 
     if (config.Events().Enabled() && !config.Offline()) {
-        event_processor_ = std::make_unique<EventProcessor>(
-            ioc_.get_executor(), config.ServiceEndpoints(), config.Events(),
-            http_properties_, logger_);
+        event_processor_ =
+            std::make_unique<events::AsioEventProcessor<ClientSDK>>(
+                ioc_.get_executor(), config.ServiceEndpoints(), config.Events(),
+                http_properties_, logger_);
```

Comment:
I've removed a now-redundant layer here. Before, `EventProcessor` was a very thin wrapper that forwarded its arguments to `AsioEventProcessor`. This worked fine when only the client SDK existed. Ideally, we could share that interface with the server.

Now that `AsioEventProcessor` implements `IEventProcessor`, the server and client SDKs can directly instantiate it instead.